cmd/go-camo/main.go

```diff
@@ -72,6 +76,12 @@ func main() {
 		os.Exit(1)
 	}
 
+	// set the server name
+	ServerName := opts.ServerName
+	if ServerCommitSha != "" && opts.ExposeGitShaInServerName {
+		ServerName = fmt.Sprintf("%s (%s)", opts.ServerName, ServerCommitSha)
```

Comment:
This could cause proxy loops, because loop detection relies on the Via header, which uses the server name.
If you were running two different versions of go-camo in a cluster, with this enabled, you could possibly expose loops.

This wouldn't be so bad, if the server name was always go-camo (prefix matching could be used), but by allowing a custom name it adds some complexity.

I think there are a few options:
1. Use `go-camo` as the Via header, (and User-Agent) regardless of what the servername is set to.

2. Do not include ServerCommitSha directly in SeverName variable, instead add it separately to router config, which handles the server response header. Then the via header would continue to only include SeverName.
 Setting the Server response header only would be done near line 175 in the master branch. 

Of those, the second option is probably the best.

Summary:
1. Do not modify ServerName to include the commit sha. 
2. Add a new variable ServerNameResponse, set to opts.ServerName. 
3. Append the ServerVersion to ServerNameResponse if a flag is set to expose the version.
4. set ServerName on dumbrouter to ServerNameResponse (instead of config.ServerName)
5. rename the flag to ExposeServerVersion `long:"expose-server-version" description:"Include the server version in the HTTP server response header"`

This should accomplish adding the version to the Server response header, but leaving ServerName unmodified for Via and User-Agent headers.

Reply:
that looks great. 👍